File: Javascript/day01/day-01-a.js
const fs = require("fs");

const text = fs.readFileSync("./input.txt", "utf-8");

let cals = text.split("\n\n").map((cals) =>
  cals
    .split("\n")
    .map((c) => parseInt(c, 10))
    .reduce((acc, el) => acc + el)
);

console.log(Math.max(...cals));

File: Go/day01/day-01-a.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	bytes, err := os.ReadFile("./input.txt")
	if err != nil {
		panic(1)
	}
	text := string(bytes)

	cals := calsToSum(strings.Split(text, "\n\n"))

	fmt.Println(cals[0])
}

func calsToSum(calories []string) []int {
	cals := make([]int, len(calories))

	for i, cal := range calories {
		sum := 0
		for _, c:= range strings.Split(cal, "\n") {
			conv, _ := strconv.Atoi(string(c))
			sum += conv
		}
		cals[i] = sum
	}

	sort.Slice(cals, func(i, j int) bool { return cals[i] > cals[j]})

	return cals
}
